pkg/resources/task_acceptance_test.go

```diff
@@ -1,564 +1,970 @@
+						HasWarehouseString(acc.TestClient().Ids.WarehouseId().Name()).
+						HasScheduleString("10 MINUTES").
+						HasConfigString(expectedTaskConfig).
+						HasAllowOverlappingExecutionString(r.BooleanTrue).
+						HasErrorIntegrationString(errorNotificationIntegration.ID().Name()).
+						HasCommentString(comment).
+						HasFinalizeString("").
+						HasNoAfter().
+						HasWhenString(condition).
+						HasSqlStatementString(statement),
+					resourceshowoutputassert.TaskShowOutput(t, configModel.ResourceReference()).
+						HasCreatedOnNotEmpty().
+						HasName(id.Name()).
+						HasIdNotEmpty().
+						HasDatabaseName(id.DatabaseName()).
+						HasSchemaName(id.SchemaName()).
+						HasOwner(currentRole.Name()).
+						HasComment(comment).
+						HasWarehouse(acc.TestClient().Ids.WarehouseId().Name()).
+						HasSchedule("10 MINUTES").
+						HasPredecessors().
+						HasState(sdk.TaskStateStarted).
+						HasDefinition(statement).
+						HasCondition(condition).
+						HasAllowOverlappingExecution(true).
+						HasErrorIntegration(errorNotificationIntegration.ID().Name()).
+						HasLastCommittedOnNotEmpty().
+						HasLastSuspendedOn("").
+						HasOwnerRoleType("ROLE").
+						HasConfig(expectedTaskConfig).
+						HasBudget("").
+						HasTaskRelations(sdk.TaskRelations{}),
+					resourceparametersassert.TaskResourceParameters(t, configModel.ResourceReference()).
+						HasAllDefaults(),
+				),
+			},
+			{
+				ResourceName: "snowflake_task.test",
+				ImportState:  true,
+				ImportStateCheck: assert.AssertThatImport(t,
+					resourceassert.ImportedTaskResource(t, helpers.EncodeResourceIdentifier(id)).
+						HasFullyQualifiedNameString(id.FullyQualifiedName()).
+						HasDatabaseString(id.DatabaseName()).
+						HasSchemaString(id.SchemaName()).
+						HasNameString(id.Name()).
+						HasEnabledString(r.BooleanTrue).
+						HasWarehouseString(acc.TestClient().Ids.WarehouseId().Name()).
+						HasScheduleString("10 MINUTES").
+						HasConfigString(expectedTaskConfig).
+						HasAllowOverlappingExecutionString(r.BooleanTrue).
+						HasErrorIntegrationString(errorNotificationIntegration.ID().Name()).
+						HasCommentString(comment).
+						HasFinalizeString("").
+						HasNoAfter().
+						HasWhenString(condition).
+						HasSqlStatementString(statement),
+				),
 			},
 		},
+	})
+}
 
-		ChildTask: &TaskSettings{
-			Name:    childname,
-			SQL:     "SELECT 1",
-			Enabled: true,
-			Comment: "third state",
-		},
+func TestAcc_Task_Updates(t *testing.T) {
+	_ = testenvs.GetOrSkipTest(t, testenvs.EnableAcceptance)
+	acc.TestAccPreCheck(t)
+
+	currentRole := acc.TestClient().Context.CurrentRole(t)
+
+	id := acc.TestClient().Ids.RandomSchemaObjectIdentifier()
+	statement := "SELECT 1"
+	basicConfigModel := model.TaskWithId("test", id, false, statement)
+
+	// New warehouse created, because the common one has lower-case letters that won't work
+	warehouse, warehouseCleanup := acc.TestClient().Warehouse.CreateWarehouse(t)
```

Comment:
We already have SNOW-1541938 for streamlits, we should create a similar ticket for this.

Reply:
+1

Reply:
Created and referenced in TODO